D96-D98 AFLUX/ADVECM/MAtoP/PGF instrumentation (ATMDYN.f part: hooks in AFLUX, ADVECM, live PGF; unit numbers are in the ATM_DRV_dynB patch). Generated by diff -u against the PRISTINE ATMDYN.f (2026-10-05); local hunks, applies to pristine or after ATMDYN_fltruv.f.patch (disjoint regions).

--- ATMDYN.f.orig
+++ ATMDYN.f
@@ -521,8 +521,15 @@
      *           USV0(IM,2,LM),VSV0(IM,2,LM), M,CONVs,MVARs
       real*8 :: mdn,mup,xx
       integer :: iup,idn,jup,jdn,adjmode,nn,j1p,j2p
+      real*8 :: ffspa0(IM,JM,LM)
 
        zNSxDT = 1 / (NS*DT)
+      call ffdb_pass_inc()
+      ffspa0(:,:,:) = 0
+      call ffdb_aflux_in(NS,IM,JM,LM,U,V,MA,MASUM,ME,MESUM,
+     &     aflux_topo_adjustments,npatch_aflux_topo_adj,
+     &     ipatch_aflux_topo_adj,jpatch_aflux_topo_adj,
+     &     md_aflux_topo_adj)
 !****                             +---------+
 !**** COMPUTATION OF MASS FLUXES  |  MA,T  MU    PRIMARY GRID ROW (J)
 !**** ARAKAWA SCHEME B            +---MV---U,V   VELOCITY GRID ROW (J)
@@ -554,6 +561,7 @@
       Do 2154 J=J1XP,JNXP
       DO 2154 I=1,IM
  2154 SPA(I,J,L)=U(I,J,L)+U(I,J+1,L)
+      ffspa0(:,:,L) = SPA(:,:,L)
       CALL AVRX (SPA(1,J1H,L))
       I=IM
       Do 2166 J=J1XP,JNXP
@@ -741,6 +749,7 @@
          If (QSP)  MW(2:IM,1 ,L) = MW(1,1 ,L)
          If (QNP)  MW(2:IM,JM,L) = MW(1,JM,L)  ;  EndDo
 
+      call ffdb_aflux_out(IM,JM,LM,ffspa0)
       RETURN
       EndSubroutine AFLUX
 
@@ -776,6 +785,7 @@
       ! 1st pass count warning/termination events
       ! This avoids the need for 2 halo fills during normal
       ! execution.
+      call ffdb_advecm_in(IM,JM,LM,DT1,MOLD)
       n_exception = 0
       Do J=J1,JN
       Do I=1,IMAXJ(J)
@@ -835,6 +845,7 @@
       Call HALO_UPDATE_COLUMN (GRID, MNEW, From=SOUTH+NORTH)
       Call HALO_UPDATE        (GRID, MSUM, From=SOUTH+NORTH)
       Call MAtoP (MNEW,MSUM)
+      call ffdb_advecm_out(IM,JM,LM,MNEW,MSUM)
       Return
   990 Format (/'0PRESSURE DIAGNOSTIC  I,J,MRCH,ZATMO,DT=',3I4,2F10.2/
      *  '  L     U(I-1,J)     U(I,J)   U(I-1,J+1)    U(I,J+1)',
@@ -1181,6 +1192,7 @@
 !**** Local variables
       Real*8,Dimension(IM,GRID%J_STRT_HALO:GRID%J_STOP_HALO,LM) ::
      *   P,PGFU
+      Real*8 :: ffpgfu0(IM,JM,LM)
       Real*8 :: DT4,DGZU(LM),DGZA(LM),
      *          M,PU,PKU,PKPU,PKPPU,DP,zDP,X,Y,PD,PKD,PKPD,PKPPD,GZD,
      *          FACTOR,FLUX, VMASS
@@ -1197,6 +1209,8 @@
      &         HAVE_SOUTH_POLE = HAVE_SOUTH_POLE,
      &         HAVE_NORTH_POLE = HAVE_NORTH_POLE)
 C****
+      ffpgfu0(:,:,:) = 0
+      call ffdb_pgf_in(IM,JM,LM,DT1,MAM,UT,VT,MAFTER,S0,SZ)
       DT4=DT1/4.
       HUNDREDTHeKAPA = .01d0**KAPA
 
@@ -1279,6 +1293,7 @@
           END DO
         END DO
 
+        ffpgfu0(:,:,L) = PGFU(:,:,L)
         Call AVRX (PGFU(1,J_0H,L), JRANGE=(/Max(2,J_0H),Min(JM-1,J_1)/))
 
         DO J=J_0STG,J_1STG
@@ -1320,6 +1335,7 @@
          VT(I,J,L) = VT(I,J,L) + DVT(I,J,L) / VMASS  ;  EndDo
  3525 I = IP1
 
+      call ffdb_pgf_out(IM,JM,LM,UT,VT,PGFU,ffpgfu0)
       RETURN
       END SUBROUTINE PGF
 #endif /* V2_PGF or not */
